bin/ec2metadata

```diff
@@ -131,10 +138,16 @@ class EC2Metadata:
 
         return False
 
+    def _imdsv2_ensure_token(self):
+        # Get IMDSv2 session token
+        request = urllib_request.Request(session_token_url, method='PUT', headers={TOKEN_HEADER_TTL: TOKEN_TTL_SECONDS})
+        resp = urllib_request.urlopen(request)
+        self.session_token = resp.read()
```

Comment:
Nit: Some linters will complain if you define instance variables outside of `__init__`. What if we name the function something like `_get_imdsv2_token` and returned the token? Then in `__init__` we can something like `self.session_token = self._get_imdsv2_token()`

Reply:
Good point @TheRealFalcon; however before moving the definition to `__init__` I'd like to understand why pylint didn't spot this. CI passes and so does a manual
```
pylint bin/ec2metadata
```
Removing the `pylintrc` and the `# pylint: disable=R0903` comment enables other warning, but not "Instance attribute attribute_name defined outside `__init__`"...

Reply:
It looks like pylint is smart enough to know if a helper is called from `__init__` and will only trigger the warning if it isn't. If you comment out line 126 you'll get the warning. So this also means you can ignore my comment. :slightly_smiling_face:

Reply:
Thanks!